tests/storage/imageupload.go

```diff
@@ -181,8 +182,11 @@ var _ = SIGDescribe("[Serial]ImageUpload", func() {
 				defer func() {
 					err = virtClient.VirtualMachineInstance(util.NamespaceTestDefault).Delete(vmi.Name, &metav1.DeleteOptions{})
 					Expect(err).ToNot(HaveOccurred())
+					tests.WaitForVirtualMachineToDisappearWithTimeout(vmi, 120)
```

Comment:
In general tests should not have to wait for a previous test to be fully cleaned up.
Normally it works like this right now:
 * In the outmost `BeforeEach`, you just call `tests.BeforeTestCleanup`. That will delete everything in the test namespace, but it will not wait for artifacts to disappear since that can take quite some time
 * You then simply create and operate on entities with random names and do your checks. The clusters have a size which allows to run at least three fedora VMs (they are the biggest VMs we use) in parallel without impacting each other.
 * Specific artifact cleanup and waiting for resources to disappear is therefore normally not necessary at all.

Reply:
I see, thanks for the elaborate explanation

Just to clarify the situation here - this test ensures DV cleanup, which in turn also ensures PVC deletion.
The problem is (at least that is what I see in [periodic](https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/logs/periodic-kubevirt-e2e-k8s-1.20-sig-storage/1437043412187484160)) that we start waiting (90 sec) on the PVC being gone before the VMI is, losing some valuable seconds and then timing out.

Reply:
Thanks for the details. The latest state of the PR seems to do something completely different now though :)

Reply:
I dropped the VMI deletion in snapshot/restore tests, that's all
Just want to start looping over PVCs being gone after we know VMI's are, to not lose any of the 90sec.